### ISSUE TITLE
Integration with elm-language-server is broken

### ISSUE THREAD

**Issue by pdamoc** [sublimelsp/LSP-elm#38]
Context: ST3 3.2.2, LSP 0.14.3, LSP-elm 1.1.7 
The analysis shows errors even if the file compiles without problems. 

Sample screen with errors for a file that compiles without problems: 
![Screenshot 2021-02-14 at 13 03 36](https://user-images.githubusercontent.com/878106/108025777-ce30a380-702f-11eb-8fba-30d1430ee2c4.png)

**Comment by predragnikolic** [comment]
This looks like a language server issue.
I would suggest reporting it here -> https://github.com/elm-tooling/elm-language-server

There is already an open issue(but don't know if that is the same issue as you are experiencing):
https://github.com/elm-tooling/elm-language-server/issues/406

**Comment by rwols** [comment]
I'm personally not experiencing this problem, but then again I don't have much experience with elm. At least this repository seems to work OK: https://github.com/stil4m/elm-analyse

**Comment by sdeframond** [comment]
Running into this bug as well. Has it been reported to https://github.com/elm-tooling/elm-language-server yet ?

I'll do it otherwise.

**Comment by sdeframond** [comment]
I feel I should add some more context...

Right after `LSP: Restart Servers`, it works just fine:
![Capture d’écran de 2021-03-24 18-16-41](https://user-images.githubusercontent.com/206747/112354895-70961380-8ccd-11eb-8360-12e823f74c16.png)
![Capture d’écran de 2021-03-24 18-19-56](https://user-images.githubusercontent.com/206747/112355042-94595980-8ccd-11eb-985e-6824416e8113.png)

But then after modifying and saving the file, the LSP gets confused and spits nonsense:
![Capture d’écran de 2021-03-24 18-17-02](https://user-images.githubusercontent.com/206747/112354935-78ee4e80-8ccd-11eb-9673-896c1ca3481f.png)

![Capture d’écran de 2021-03-24 18-17-14](https://user-images.githubusercontent.com/206747/112354963-7ee42f80-8ccd-11eb-9d91-ac2b962d4017.png)

Restarting LSP Servers again temporarily fixes the issue, until next change.

**Comment by cbossut** [comment]
Is there anything new regarding this problem ?
This currently makes the setup Sublime + LSP-elm totally unusable.
I found related issues [here](https://github.com/elm-tooling/elm-language-server/issues/598), [here](https://github.com/elm-tooling/elm-language-server/issues/563) and [here](https://github.com/elm-tooling/elm-language-server/issues/503), but cannot understand everything, and nobody is mentionning ST in these.

**Comment by predragnikolic** [comment]
That problem seems to be fixed in elm language server V2.2.0. 
Currently LSP-elm uses v2.2.1 but if you are still experiencing the bug, I could take a look.

**Comment by predragnikolic** [comment]
Lets sum up.

There was [an issue](https://github.com/elm-tooling/elm-language-server/issues/635) on the server that made the server not work correctly for clients that do not support incremental changes. This was fixed with https://github.com/elm-tooling/elm-language-server/pull/642

This issue was reported for Sublime Text 3.
I do not remember if LSP for ST 3 supports incremental changes. 
If I recall correctly incremental changes were added for LSP in ST 4. (thus I could not replicate the bug. ) 

I made a new release elm that uses the new elm server v.2.3.0 that include the fix. 

I will close this, 
but if you still experience some issue feel free to open them at the elm language server repo.
